client-src/elements/chromedash-feature-page.ts

```diff
@@ -119,6 +127,56 @@ export class ChromedashFeaturePage extends LitElement {
     return this.feature && Object.keys(this.feature).length !== 0;
   }
 
+  isFeatureUpcoming(channels, stages: Array<StageDict>) {
+    const latestStableVersion = channels['stable']?.version;
+    if (!latestStableVersion || !stages) {
+      return;
+    }
+
+    const shippingMilestones = new Set<number | undefined>();
+    // Get milestones from all shipping stages, STAGE_TYPES_SHIPPING.
+    for (const stage of stages) {
+      if (STAGE_TYPES_SHIPPING.has(stage.stage_type)) {
+        shippingMilestones.add(stage.desktop_first);
+        shippingMilestones.add(stage.android_first);
+        shippingMilestones.add(stage.ios_first);
+        shippingMilestones.add(stage.webview_first);
+      }
+    }
+    // Check if this feature is shipped within two milestones.
+    let foundMilestone = 0;
+    if (shippingMilestones.has(latestStableVersion + 1)) {
+      foundMilestone = latestStableVersion + 1;
+      this.isUpcoming = true;
+    } else if (shippingMilestones.has(latestStableVersion + 2)) {
+      foundMilestone = latestStableVersion + 2;
+      this.isUpcoming = true;
+    }
+
+    if (this.isUpcoming) {
+      Object.keys(channels).forEach(key => {
+        if (channels[key].version === foundMilestone) {
+          this.closestShippingMilestone = channels[key].final_beta;
```

Comment:
I think this would be more clear if the property was named `this.closestShippingDate`.